examples/reference-entity-model/entities/Customer.entity.json

```diff
@@ -1,98 +1,111 @@
-          }
-        },
-        "city": {
-          "datatype": "string",
-          "collation": "http://marklogic.com/collation/codepoint"
-        },
-        "state": {
-          "datatype": "string",
-          "collation": "http://marklogic.com/collation/codepoint"
-        },
-        "zip": {
-          "$ref": "#/definitions/Zip"
-        }
-      }
-    },
-    "Zip": {
-      "required": [ ],
-      "properties": {
-        "fiveDigit": {
-          "datatype": "string",
-          "collation": "http://marklogic.com/collation/codepoint"
-        },
-        "plusFour": {
-          "datatype": "string",
-          "collation": "http://marklogic.com/collation/codepoint"
-        }
-      }
-    }
-  }
-}
+{
+  "info": {
+    "title": "Customer",
+    "version": "0.0.1",
+    "baseUri": "http://example.org/"
+  },
+  "definitions": {
+    "Customer": {
+      "required": [
+        "name"
+      ],
+      "primaryKey": "customerId",
+      "properties": {
+        "customerId": {
+          "datatype": "integer",
+          "facet": true
+        },
+        "name": {
+          "datatype": "string",
+          "facet": false,
+          "collation": "http://marklogic.com/collation/codepoint"
+        },
+        "nicknames": {
+          "datatype": "array",
+          "description": "Example of a multi-value property of simple values",
+          "facet": false,
+          "items": {
+            "datatype": "string"
+          }
+        },
+        "shipping": {
+          "datatype": "array",
+          "description": "Example of a multi-value property of structured values",
+          "facet": false,
+          "items": {
+            "$ref": "#/definitions/Address"
+          }
+        },
+        "billing": {
+          "description": "Example of a single-value structured property",
+          "$ref": "#/definitions/Address"
+        },
+        "birthDate": {
+          "datatype": "date",
+          "facet": true
+        },
+        "status": {
+          "datatype": "string",
+          "facet": false
+        },
+        "customerSince": {
+          "datatype": "date",
+          "facet": false
+        },
+        "orders": {
+          "datatype": "array",
+          "description": "Example of a relationship to another entity type",
+          "facet": false,
+          "items": {
+            "$ref": "http://example.org/Order-0.0.1/Order"
+          }
+        }
+      }
+    },
+    "Address": {
+      "required": [ ],
+      "pii": [ ],
+      "elementRangeIndex": [ ],
+      "rangeIndex": [ ],
+      "wordLexicon": [ ],
+      "properties": {
+        "street": {
+          "datatype": "array",
+          "facet": false,
+          "items": {
+            "datatype": "string",
+            "collation": "http://marklogic.com/collation/codepoint"
+          }
+        },
+        "city": {
+          "datatype": "string",
+          "facet": true,

```

Comment:
We're not going to support facets on structured properties for this first go-round.